clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -70,6 +75,17 @@ public AbstractConfig(ConfigDef definition, Map<?, ?> originals) {
         this(definition, originals, true);
     }
 
+    /**
+     * Called directly after user configs got parsed (and thus default values got set).
+     * This allows to change default values for "secondary defaults" if required.
+     *
+     * @param parsedValues unmodifiable map of current configuration
+     * @return a map of updates that should be applied to the configuration (will be validated to prevent bad updates)
+     */
+    protected Map<String, Object> postProcessParsedConfig(Map<String, Object> parsedValues) {
```

Comment:
In the initial PR, I did not modify `AbstractConfig` and did everything in `StreamsConfig`. @ijuma did not like this and requested to do it in `AbstractConfig`. In the first version, I just made `this.originals` protected so I can modify the default value -- but this was the disadvantage that `originals` is not immutable anymore. Thus, we come up with this solution.

Personally, I am fine either way. Looking for further feedback before changing forth-and-back again.